targets/atlys_vga.py

```diff
@@ -0,0 +1,241 @@
+        "hdmi_out0": 19,
+    }
+    csr_map.update(SDRAMSoC.csr_map)
+
+    interrupt_map = {
+        "vga_in": 2,
+    }
+    interrupt_map.update(SDRAMSoC.interrupt_map)
+
+    mem_map = {
+        "firmware_ram": 0x20000000,  # (default shadow @0xa0000000)
+    }
+    mem_map.update(SDRAMSoC.mem_map)
+
+    def __init__(self, platform,
+                 firmware_ram_size=0xa000,
+                 firmware_filename=None,
+                 **kwargs):
+        clk_freq = 75*1000000
+        SDRAMSoC.__init__(self, platform, clk_freq,
+                          integrated_rom_size=0x8000,
+                          sdram_controller_settings=LASMIconSettings(l2_size=32, with_bandwidth=True),
+                          **kwargs)
+
+        self.submodules.crg = _CRG(platform, clk_freq)
+
+        self.submodules.firmware_ram = wishbone.SRAM(firmware_ram_size, init=_get_firmware_data(firmware_filename))
+        self.register_mem("firmware_ram", self.mem_map["firmware_ram"], self.firmware_ram.bus, firmware_ram_size)
+        self.add_constant("ROM_BOOT_ADDRESS", self.mem_map["firmware_ram"])
+
+        if not self.integrated_main_ram_size:
+            self.submodules.ddrphy = s6ddrphy.S6HalfRateDDRPHY(platform.request("ddram"),
+                                                               P3R1GE4JGF(self.clk_freq),
+                                                               rd_bitslip=0,
+                                                               wr_bitslip=4,
+                                                               dqs_ddr_alignment="C0")
+            self.comb += [
+                self.ddrphy.clk4x_wr_strb.eq(self.crg.clk4x_wr_strb),
+                self.ddrphy.clk4x_rd_strb.eq(self.crg.clk4x_rd_strb),
+            ]
+            self.register_sdram_phy(self.ddrphy)
+
+        self.specials += Keep(self.crg.cd_sys.clk)
+                
+        vga_pads = platform.request("vga", 0)
+        self.submodules.vga_in = VGAIn(vga_pads,
+                                       self.sdram.crossbar.get_master(),
+                                       fifo_depth=1024)
+        self.submodules.i2c = I2C(vga_pads)
+
+        self.submodules.hdmi_out0 = HDMIOut(platform.request("hdmi_out", 0),
+                                            self.sdram.crossbar.get_master())
+        platform.add_platform_command("""PIN "hdmi_out_pix_bufg.O" CLOCK_DEDICATED_ROUTE = FALSE;""")
+        platform.add_platform_command("""
+NET "{pix0_clk}" TNM_NET = "GRPpix0_clk";
+TIMESPEC "TSise_sucks7" = FROM "GRPpix0_clk" TO "GRPsys_clk" TIG;
+TIMESPEC "TSise_sucks8" = FROM "GRPsys_clk" TO "GRPpix0_clk" TIG;
+TIMESPEC "TSise_sucks9" = FROM "GRPpix1_clk" TO "GRPsys_clk" TIG;
+TIMESPEC "TSise_sucks10" = FROM "GRPsys_clk" TO "GRPpix1_clk" TIG;
+""", pix0_clk=self.hdmi_out0.driver.clocking.cd_pix.clk,
+)
+        
+        platform.add_platform_command("""
+NET "{sys_clk}" TNM_NET = "GRPsys_clk";
+""", sys_clk=self.crg.cd_sys.clk)
+
+
+class MiniSoC(BaseSoC):
```

Comment:
I thought that the having both Ethernet and the VGA board running at the same time wasn't possible due to IO level incompatibilities?